cmd/dra-example-kubeletplugin/driver.go

```diff
@@ -49,87 +49,103 @@ func NewDriver(ctx context.Context, config *Config) (*driver, error) {
 
-	plugin, err := kubeletplugin.Start(
+	helper, err := kubeletplugin.Start(
 		ctx,
-		[]any{driver},
+		driver,
 		kubeletplugin.KubeClient(config.coreclient),
 		kubeletplugin.NodeName(config.flags.nodeName),
 		kubeletplugin.DriverName(consts.DriverName),
-		kubeletplugin.RegistrarSocketPath(PluginRegistrationPath),
-		kubeletplugin.PluginSocketPath(DriverPluginSocketPath),
-		kubeletplugin.KubeletPluginSocketPath(DriverPluginSocketPath))
+	)
 	if err != nil {
 		return nil, err
 	}
-	driver.plugin = plugin
+	driver.helper = helper
 
-	var resources kubeletplugin.Resources
-	for _, device := range state.allocatable {
-		resources.Devices = append(resources.Devices, device)
+	devices := make([]resourceapi.Device, 0, len(state.allocatable))
+	for device := range maps.Values(state.allocatable) {
+		devices = append(devices, device)
+	}
+	resources := resourceslice.DriverResources{
+		Pools: map[string]resourceslice.Pool{
+			config.flags.nodeName: {
+				Slices: []resourceslice.Slice{
+					{
+						Devices: devices,
+					},
+				},
+			},
+		},
 	}
 
-	if err := plugin.PublishResources(ctx, resources); err != nil {
+	if err := helper.PublishResources(ctx, resources); err != nil {
 		return nil, err
 	}
 
 	return driver, nil
 }
 
 func (d *driver) Shutdown(ctx context.Context) error {
-	d.plugin.Stop()
+	d.helper.Stop()
 	return nil
 }
 
-func (d *driver) NodePrepareResources(ctx context.Context, req *drapbv1.NodePrepareResourcesRequest) (*drapbv1.NodePrepareResourcesResponse, error) {
-	klog.Infof("NodePrepareResource is called: number of claims: %d", len(req.Claims))
-	preparedResources := &drapbv1.NodePrepareResourcesResponse{Claims: map[string]*drapbv1.NodePrepareResourceResponse{}}
+func (d *driver) PrepareResourceClaims(ctx context.Context, claims []*resourceapi.ResourceClaim) (map[types.UID]kubeletplugin.PrepareResult, error) {
+	klog.Infof("PrepareResourceClaims is called: number of claims: %d", len(claims))
+	result := make(map[types.UID]kubeletplugin.PrepareResult)
 
-	for _, claim := range req.Claims {
-		preparedResources.Claims[claim.UID] = d.nodePrepareResource(ctx, claim)
+	for _, claim := range claims {
+		result[claim.UID] = d.prepareResourceClaim(ctx, claim)
 	}
 
-	return preparedResources, nil
+	return result, nil
 }
 
-func (d *driver) nodePrepareResource(ctx context.Context, claim *drapbv1.Claim) *drapbv1.NodePrepareResourceResponse {
+func (d *driver) prepareResourceClaim(ctx context.Context, claim *resourceapi.ResourceClaim) kubeletplugin.PrepareResult {
 	resourceClaim, err := d.client.ResourceV1beta1().ResourceClaims(claim.Namespace).Get(
 		ctx,
 		claim.Name,
 		metav1.GetOptions{})
 	if err != nil {
-		return &drapbv1.NodePrepareResourceResponse{
-			Error: fmt.Sprintf("failed to fetch ResourceClaim %s in namespace %s", claim.Name, claim.Namespace),
+		return kubeletplugin.PrepareResult{
+			Err: fmt.Errorf("failed to fetch ResourceClaim %s in namespace %s: %w", claim.Name, claim.Namespace, err),
 		}
 	}
```

Comment:
Do we still need this? The helper ensures presence of claim in the given namespace.
- https://github.com/kubernetes/kubernetes/blob/b53b9fb5573323484af9a19cf3f5bfe80760abba/staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go#L731

Reply:
It looks like we don't. I missed that the helper already does this for us. Thanks!